watchdog_linux.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/coreos/go-systemd/daemon"
+	"github.com/muesli/beehive/api"
+	log "github.com/sirupsen/logrus"
+)
+
+// Notifies Systemd's watchdog every WatchdogSec/3 when running
+// under Systemd and the watchdog feature has been enabled in
+// the service unit.
+//
+// This will no-op when not running under Systemd.
+//
+// See http://0pointer.de/blog/projects/watchdog.html
+// and https://www.freedesktop.org/software/systemd/man/systemd.service.html
+//
+func init() {
+	interval, err := daemon.SdWatchdogEnabled(false)
+	// systemd's service unit interval in microseconds
+	if err != nil || interval == 0 {
+		log.Printf("Systemd watchdog not enabled")
+		return
+	}
+	interval = interval / 3 / 1000000000
```

Comment:
I'm not quite sure I understand that line. Between what units are we converting here?

Reply:
Interval is microseconds according to the docs, while WatchdogSec is is seconds, so a  value of 30 secs in the systemd unit would be 30000000/3/1000000, or notify every 10 secs. Does that make sense? Sound like I should add some more comments.

Reply:
Now that you mention it, I’ve added some extras that should be there. Let me have a closer look.

Reply:
I see. We could probably use Go's `time.Millisecond` / `time.Second` to make it a bit more readable.

Reply:
The math was correct buy I made it overly  confusing because I still do not understand how Golang's time.Duration works.